### PR TITLE
[#163] Added reset password and email notification on mail with new password

### PR DESCRIPTION
**Pull request by laganows**
Based on SAT of course :)

### CODE REVIEW

File: src/main/java/com/beerjournal/breweriana/exchange/ExchangeUpdateService.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ ExchangeOfferDetailsDto updateExchange(String exchangeId, ExchangeUpdateDto upda
 
     private void validateModifiability(ExchangeState state) {
         if(!state.isModifiable()) {
-            throw new BeerJournalException(ErrorInfo.EXCHANGE_NOT_MODIFIABLE);
+            throw new BeerJournalException(ErrorInfo.MAIL_NOT_SEND);
```
Comment (pkruczek) [inline src/main/java/com/beerjournal/breweriana/exchange/ExchangeUpdateService.java]: MAIL_NOT_SEND?

Comment (laganows) [reply src/main/java/com/beerjournal/breweriana/exchange/ExchangeUpdateService.java]: My bad, fixed :)

File: src/main/java/com/beerjournal/infrastructure/error/ErrorInfo.java [hunk 2]
```diff
@@ -31,14 +32,17 @@
...
+    // Exchanges
     EXCHANGE_NOT_FOUND(4104, "Exchange not found", HttpStatus.NOT_FOUND),
     EXCHANGE_FORBIDDEN_MODIFICATION(4105, "You do not have permission to modify that exchange", HttpStatus.FORBIDDEN),
     MALFORMED_STATE(4106, "Bad offer state name", HttpStatus.BAD_REQUEST),
     EXCHANGE_FORBIDDEN_STATE_MODIFICATION(4107, "You do not have permission to modify exchange state", HttpStatus.FORBIDDEN),
     EXCHANGE_BAD_NEXT_STATE(4108, "Bad next exchange state", HttpStatus.CONFLICT),
     EXCHANGE_NOT_MODIFIABLE(4109, "Given offer is not midifiable", HttpStatus.CONFLICT),
 
+    // Mail
+    MAIL_NOT_SEND(4200, "Mail was not sent", HttpStatus.CONFLICT),
```
Comment (pkruczek) [inline src/main/java/com/beerjournal/infrastructure/error/ErrorInfo.java]: maybe MAIL_NOT_SENT

Comment (laganows) [reply src/main/java/com/beerjournal/infrastructure/error/ErrorInfo.java]: Still better than D.T. ^^ Ok, changed.

**Review by cwikp — APPROVED** [review]

**Pull request merged**